### PR TITLE
Andrewrutter/kkb 38 delete event backend

### PR DESCRIPTION
**Pull request by andlrutt**
Created `deleteEvent()` function — which takes an ID as a parameter and deletes the corresponding event from the db — the api route for it, and tests. Tested in postman creating an entry in the db and sending `localhost:3000/api/events/[eventId]` with the `DELETE` method, which successfully removed that entry from the db. 
I am a little shaky on the tests, so if you could look at those with a bit more scrutiny than usual, it would be much appreciated!

Also, the `deleteEvent()` function simply returns the payload `success: true` and nothing else, so if you'd like something else in there, like a confirmation message, or the ID of the deleted entry, please let me know.

### CODE REVIEW

File: tst/server/Event.test.ts [hunk 1]
```diff
@@ -71,3 +71,38 @@ describe("addEvent() tests", () => {
...
+    test("no event with that id", async () => {
+        expect.assertions(1);
+        // mongoose returns undefined when a query results in no results
+        const mockEvent = undefined;
+
+        //mock EventSchema.findById to return mockEvent, which will then throw an error
+        EventSchema.findById = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
+    });
+
```
Comment (kemalfidan) [inline tst/server/Event.test.ts]: You can remove these tests. These seem to be testing getEvent, so deleteEvent doesn't need these.

File: tst/server/Event.test.ts [hunk 2]
```diff
@@ -71,3 +71,38 @@ describe("addEvent() tests", () => {
...
+    test("event deleted", async () => {
+        const mockEvent = {
+            name: "test",
+        };
+
+        EventSchema.findByIdAndDelete = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
+    });
+});
```
Comment (kemalfidan) [inline tst/server/Event.test.ts:92]: Can you add another test that runs a failing case? Like if the id is ""?        
Something like `await expect(deleteEvent("")).rejects.toThrowError("Invalid id");`

File: tst/server/Event.test.ts [hunk 3]
```diff
@@ -71,3 +71,38 @@ describe("addEvent() tests", () => {
...
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
+    });
+
+    test("event deleted", async () => {
+        const mockEvent = {
+            name: "test",
+        };
+
+        EventSchema.findByIdAndDelete = jest.fn().mockResolvedValue(mockEvent);
+        await expect(getEvent("602734007d7de15fae321152")).rejects.toThrowError("Event does not exist");
```
Comment (kemalfidan) [inline tst/server/Event.test.ts]: What this line does is it runs `getEvent()` with a certain ID and expects to reject the promise returned by the function. What you need to do is run `deleteEvent()` and make sure that `EventSchema.findByIdAndDelete` was ran with certain parameters. 

You can do this by mocking the implementation of `findByIdAndDelete` with the following: `EventSchema.findByIdAndDelete = jest.fn().mockImplementation(async (event: Event) => event);`. Then you can run the function and have checks to make sure things worked correctly:
`await addEvent(mockEvent);`
`expect(EventSchema.findByIdAndDelete).lastCalledWith(mockEvent);`
`expect(EventSchema.findByIdAndDelete).toHaveBeenCalledTimes(1);`

**Review by kemalfidan — CHANGES_REQUESTED** [review]

File: tst/server/Event.test.ts (hunk 3 above)
Comment (andlrutt) [reply tst/server/Event.test.ts]: That makes a lot of sense, thank you!

**Review by kemalfidan — APPROVED** [review]
Looks great!

**Review by zaviermiller — APPROVED** [review]
wtg on testing!

**Review by rossketron — APPROVED** [review]
Looks great!

**Pull request merged**